core/domain/opportunity_services.py

```diff
@@ -0,0 +1,370 @@
+        complete_translation_language_list: list(str). A list of language code
+            in which the exploration translation is 100%.
+    """
+    model = opportunity_models.ExplorationOpportunitySummaryModel.get(exp_id)
+    exploration_opportunity_summary = (
+        get_exploration_opportunity_summary_from_model(model))
+    exploration_opportunity_summary.content_count = content_count
+    exploration_opportunity_summary.translation_counts = translation_counts
+    exploration_opportunity_summary.complete_translation_lanuages = (
+        complete_translation_language_list)
+
+    new_languages_for_voiceover = set(complete_translation_language_list) - set(
+        exploration_opportunity_summary.assigned_voiceartist_in_languages)
+
+    # We only append new languages to need_voiceartist_in_languages, as the
+    # complete translation languages list will be dynamic based on some
+    # content text are changed, where as the voiceover is a long term work and
+    # we can allow a voiceartist to work for an exploration which needs a little
+    # bit update in text translation.
+    need_voiceartist_in_languages_set = set(
+        exploration_opportunity_summary.need_voiceartist_in_languages)
+    need_voiceartist_in_languages_set |= set(new_languages_for_voiceover)
+
+    exploration_opportunity_summary.need_voiceartist_in_languages = list(
+        need_voiceartist_in_languages_set)
+
+    exploration_opportunity_summary.validate()
+
+    _save_multi_exploration_opportunity_summary(
+        [exploration_opportunity_summary])
+
+
+def update_exploration_opportunities_with_new_story(story, exp_ids):
+    """Updates the opportunities models with the new story.
+
+    Args:
+        story: Story. The new story object.
+        exp_ids: list(str). A list of exploration ids. Corresponding to these
+            ids exploration opportunity summary models to be updated.
+    """
+    exp_opportunity_models = (
+        opportunity_models.ExplorationOpportunitySummaryModel.get_multi(
+            exp_ids))
+
+    exploration_opportunity_summary_list = []
+
+    for exp_opportunity_model in exp_opportunity_models:
+        exploration_opportunity_summary = (
+            get_exploration_opportunity_summary_from_model(
+                exp_opportunity_model))
+        exploration_opportunity_summary.story_title = story.title
+        node = story.story_contents.get_node_with_corresponding_exp_id(
+            exploration_opportunity_summary.id)
+        exploration_opportunity_summary.chapter_title = node.title
+        exploration_opportunity_summary.validate()
+
+        exploration_opportunity_summary_list.append(
+            exploration_opportunity_summary)
+
+    _save_multi_exploration_opportunity_summary(
+        exploration_opportunity_summary_list)
+
+
+def delete_exploration_opportunities(exp_ids):
+    """Deletes the exploration_ooportunity_summary models.
```

Comment:
Deletes the XXX models corresponding to the given exp_ids.

Also, opportunity is misspelled. Please proofread your work.